### PR TITLE
Anko - Update Logic Home Screen

### PR DESCRIPTION
**Pull request by at-toannguyen1**
#### Related issues (Required)
- Task #140 
- Bug #

#### Description (Required)
- [x] update logic home screen

### CODE REVIEW

File: app/src/main/java/vn/asiantech/way/ui/custom/FloatingMenuButton.kt [hunk 1]
```diff
@@ -182,39 +187,27 @@ class FloatingMenuButton(private var onMenuClickListener: OnMenuClickListener,
...
+                rlProfile.visibility == View.INVISIBLE ||
+                rlCalendar.visibility == View.INVISIBLE ||
+                rlSearch.visibility == View.INVISIBLE ||
+                rlGroup.visibility == View.INVISIBLE
+    }
+
+    private fun setGoneOverLay() {
+        frOverlay.visibility = View.GONE
+    }
+
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/custom/FloatingMenuButton.kt]: WARNING:

File: app/src/main/java/vn/asiantech/way/ui/custom/FloatingMenuButton.kt [hunk 2]
```diff
@@ -182,39 +187,27 @@ class FloatingMenuButton(private var onMenuClickListener: OnMenuClickListener,
         rlGroup.visibility = visibilityState
     }
 
+    private fun checkItemViewVisibility(): Boolean {
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/custom/FloatingMenuButton.kt]: WARNING:

File: app/src/main/java/vn/asiantech/way/ui/custom/FloatingMenuButton.kt [hunk 3]
```diff
@@ -25,81 +23,89 @@ import vn.asiantech.way.R
...
+                        padding = dimen(R.dimen.padding_floating)
+                    }
+                }.applyRecursively { view: View ->
+                    when (view) {
+                        is RelativeLayout -> {
+                            view.gravity = Gravity.END
+                            view.visibility = View.INVISIBLE
+                        }
+                        is TextView -> view.gravity = Gravity.CENTER
+                        is ImageButton -> {
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/custom/FloatingMenuButton.kt]: WARNING: Optional braces in when expression

File: app/src/main/java/vn/asiantech/way/ui/home/HomeAdapter.kt [hunk 4]
```diff
@@ -171,12 +172,14 @@ class HomeAdapter(private val context: Context,
...
+    /**
+     * Function to custom expandableTextView
+     */
+    inline fun ViewManager.expandableTextView(init: ExpandableTextView.() -> Unit)
+            : ExpandableTextView {
+        return ankoView({ ExpandableTextView(it) }, theme = 0, init = init)
+    }
 }
+
+
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/home/HomeAdapter.kt]: WARNING:

File: app/src/main/java/vn/asiantech/way/ui/home/HomeAdapter.kt [hunk 5]
```diff
@@ -171,12 +172,14 @@ class HomeAdapter(private val context: Context,
...
- * Function to custom expandableTextView
- */
-inline fun ViewManager.expandableTextView(init: ExpandableTextView.() -> Unit)
-        : ExpandableTextView {
-    return ankoView({ ExpandableTextView(it) }, theme = 0, init = init)
+    /**
+     * Function to custom expandableTextView
+     */
+    inline fun ViewManager.expandableTextView(init: ExpandableTextView.() -> Unit)
+            : ExpandableTextView {
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/home/HomeAdapter.kt]: WARNING:

File: app/src/main/java/vn/asiantech/way/ui/home/HomeViewModel.kt [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+    }
+
+    private fun handleBackKeyEvent(): Boolean {
+        if (SystemClock.elapsedRealtime() - lastClickTime < AppConstants.TIME_EXIT_APP) {
+            return true
+        }
+        lastClickTime = SystemClock.elapsedRealtime()
+        return false
+    }
+}
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/home/HomeViewModel.kt]: WARNING: Checks whether files end with a line separator.

File: app/src/main/java/vn/asiantech/way/ui/custom/FloatingMenuButton.kt [hunk 7]
```diff
@@ -182,39 +185,27 @@ class FloatingMenuButton(private var onMenuClickListener: OnMenuClickListener,
...
+                rlProfile.visibility == View.INVISIBLE ||
+                rlCalendar.visibility == View.INVISIBLE ||
+                rlSearch.visibility == View.INVISIBLE ||
+                rlGroup.visibility == View.INVISIBLE
+    }
+
+    private fun setGoneOverLay() {
+        frOverlay.visibility = View.GONE
+    }
+
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/way/ui/custom/FloatingMenuButton.kt]: remove this line

File: app/src/main/java/vn/asiantech/way/ui/home/HomeActivity.kt [hunk 8]
```diff
@@ -1,84 +1,109 @@
...
         private const val UNIT_PADDING_BOTTOM = 3
     }
 
-    private var position = -1
-    private var googleMap: GoogleMap? = null
     private lateinit var homeAdapter: HomeAdapter
     private lateinit var ui: HomeActivityUI
+    private lateinit var homeViewModel: HomeViewModel
+
+    private var locations: MutableList<TrackingInformation> = mutableListOf()
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/way/ui/home/HomeActivity.kt]: change this line to `private var locations = mutableListOf<TrackingInformation>()`

File: app/src/main/java/vn/asiantech/way/ui/custom/FloatingMenuButton.kt [hunk 9]
```diff
@@ -182,39 +175,26 @@ class FloatingMenuButton(private var onMenuClickListener: OnMenuClickListener,
         rlGroup.visibility = visibilityState
     }
 
+    private fun checkItemViewVisibility(): Boolean {
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/custom/FloatingMenuButton.kt]: WARNING:

File: app/src/main/java/vn/asiantech/way/ui/home/HomeAdapter.kt [hunk 10]
```diff
@@ -171,12 +172,12 @@ class HomeAdapter(private val context: Context,
...
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/home/HomeAdapter.kt]: WARNING:

**Comment by at-vinhhuynh** [comment]
@at-tienhoang 
I finish review this PR, please check it again.
thank you!

File: app/src/main/java/vn/asiantech/way/ui/home/HomeActivity.kt [hunk 11]
```diff
@@ -1,84 +1,109 @@
...
     companion object {
+        private const val ZOOM = 16f
         private const val UNIT_PADDING_BOTTOM = 3
     }
 
-    private var position = -1
-    private var googleMap: GoogleMap? = null
     private lateinit var homeAdapter: HomeAdapter
     private lateinit var ui: HomeActivityUI
+    private lateinit var homeViewModel: HomeViewModel
```
Comment (at-tienhoang) [inline app/src/main/java/vn/asiantech/way/ui/home/HomeActivity.kt]: Rename to `viewModel`

**Pull request merged**